owncloudApp/src/main/java/com/owncloud/android/authentication/AccountAuthenticator.java

```diff
@@ -306,69 +305,87 @@ private boolean canBeRefreshed(String authTokenType) {
     }
 
-    private String refreshToken(Account account, String authTokenType, AccountManager accountManager) {
-
-        Timber.v("Refreshing token!");
-        String accessToken;
-        try {
-            String refreshToken = accountManager.getUserData(
-                    account,
-                    AccountUtils.Constants.KEY_OAUTH2_REFRESH_TOKEN
-            );
-
-            if (refreshToken == null || refreshToken.length() <= 0) {
-                Timber.w("No refresh token stored for silent renewal of access token");
-                return null;
-            }
-
-            OAuth2Provider oAuth2Provider = OAuth2ProvidersRegistry.getInstance().getProvider();
-
-            OAuth2RequestBuilder builder = oAuth2Provider.getOperationBuilder();
-            builder.setRequest(OAuth2RequestBuilder.OAuthRequest.REFRESH_ACCESS_TOKEN);
-            builder.setRefreshToken(refreshToken);
-
-            OAuth2RefreshAccessTokenOperation oAuth2RefreshAccessTokenOperation =
-                    (OAuth2RefreshAccessTokenOperation) builder.buildOperation();
+    private void refreshToken(Account account, String authTokenType, AccountManager accountManager,
+                              AccountAuthenticatorResponse accountAuthenticatorResponse) {
 
-            OwnCloudClient client = OwnCloudClientFactory.createOwnCloudClient(
-                    Uri.parse(accountManager.getUserData(account, AccountUtils.Constants.KEY_OC_BASE_URL)),
-                    mContext,
-                    true
-            );
+        // Prepare everything to perform the token request
 
-            RemoteOperationResult<Map<String, String>> result = oAuth2RefreshAccessTokenOperation.execute(client);
-            if (!result.isSuccess()) {
-                Timber.e("Failed to refresh access token");
-                return null;
-            }
-
-            // Get new access and refresh tokens
-            Map<String, String> tokens = result.getData();
-
-            accessToken = tokens.get(OAuth2Constants.KEY_ACCESS_TOKEN);
-
-            Timber.d("Got OAuth2 access token: %s", accessToken);
-
-            accountManager.setAuthToken(account, authTokenType, accessToken);
+        String refreshToken = accountManager.getUserData(
+                account,
+                AccountUtils.Constants.KEY_OAUTH2_REFRESH_TOKEN
+        );
 
-            Timber.d("Set OAuth2 new access token in account: %s", accessToken);
-
-            refreshToken = tokens.get(OAuth2Constants.KEY_REFRESH_TOKEN);
-
-            Timber.d("Got OAuth2 refresh token: %s", refreshToken);
-
-            accountManager.setUserData(
-                    account,
-                    AccountUtils.Constants.KEY_OAUTH2_REFRESH_TOKEN,
-                    refreshToken
-            );
-
-        } catch (Exception e) {
-            Timber.e(e, "Failed to store refreshed access token");
-            return null;
+        if (refreshToken == null || refreshToken.length() <= 0) {
```

Comment:
I did not change it because `refreshToken.isEmpty()` is not the same as `refreshToken.length() <= 0` since it does not check whether the length is <0. I kept using the code we had before, maybe refreshToken could be initialized to -1 somewhere before? 

Since now we retrieve the refreshToken from a different place (AppAuth) I will change it to `refreshToken.isEmpty() `